### ISSUE TITLE
solution proposal to not-yet-resolved part of #186

### ISSUE THREAD

**Issue by dettwild** [croneter/PlexKodiConnect#190]
First, thank you for implementing the 'Limit PMS Items' parameter so quickly. I noticed that you seem to read the limit value directly from the settings window when generating the relevant video nodes and then store it as parameter in the respective video node. 
Could it make sense to persists that value in the settings file, not storing it in the video nodes at all, but instead having the code in entrypoint,py pick up the value from settings before applying it (similar to how this is already done with settings like 'OnDeckTVextended')?
I think that this approach would result in the advantage that a change of that value by the user would no longer require a re-build of the nodes and could just be applied automatically at the next refresh.
Thoughts or comments?

Now let me get to my findings regarding the remaining issue(s) from #186:

**Part1: number of Recent Episodes shown:**
You may remember that I mentioned that the actual number of items shown in the Recent Episodes widget can be less (I said by 1, but that turned out to be inaccurate - see below) than the given limit. The answer lies in line 714 of entrypoint,py. The underlying query is just an intermediate step in obtaining the tagname-filtered recent episodes - if one already limits the results of the query at this point, it can be the case that the query also returns episodes that belong to shows with the 'wrong' tagname and therefore those episodes would drop out a few lines below when the comparison to the pre-filtered TV Show list is done.
So as a "quick fix" I would propose to remove the limit from the query.
This may have a negative impact on the performance though, given that the query needs to return ALL episodes at first (almost 5k in my case), only to then pick the first 'handfull' (or whatever the user set as his limit) - however, it does work.
Having said this I started wondering, if it would not be easier and more efficient to pull that list directly from Plex, the same way like it is done for 'OnDeck' already. If you are interested I can post the respective code that I modified quickly - I am still trying to compare performances at this point though.

**Part 2: number of 'On Deck' items (TV Shows) shown:**
Here it turns out that the code in entrypoint.py (lines 950 to 1004) does NOT take account of the limit at all. It is relatively easy to add the counter variable and break out of the loop when the counter reaches the limit.
Quick comment/question here: If a user opts to extend his On Deck view to all TV Shows, the resulting view will be significantly different from what one sees in Plex and I wonder why this option even exists given that PKC openly assumes that a user manages all his media in Plex. Also, if someone wants to influence what he sees On Deck in Plex, there is a Plex setting called "Weeks to consider for On Deck" under Server-Library. Not a criticism of course, just an observation that I am happy to discuss;-).

PS: Sorry for the lengthiness of this post.

**Comment by croneter** [comment]
Thanks! 

For next time, to prevent you having to write lengthy issues ;-), have a look at this http://hisham.hm/2016/01/01/how-to-make-a-pull-request-on-github-a-quick-tutorial/

**Comment by dettwild** [comment]
Yup, I was considering a PR, but didn't quite "dare" yet ;-). Thanks for the tutorial though, I guess I got some "homework" to do :-p.
For the time being, perhaps I could edit some simple changes directly in my forked repo and PR from there  if you are ok with that. 

Any thoughts already on obtaining the Recent Episodes directly from Plex rather than from the Kodi db?
Btw, when comparing the 2 sources I noticed (using the Titan Beta Skin) that they have different Posters. Plex displays the TV Show level poster for all recently added episodes, while Kodi (via PKC videonode) shows the Season-level Poster for those Episodes. I am nut sure yet where that diff comes from, suspect it is a native difference between Plex and Kodi.
